scene/resources/audio_stream_wav.cpp

```diff
@@ -189,7 +189,7 @@ void AudioStreamPlaybackWAV::do_resample(const Depth *p_src, AudioFrame *p_dst,
 							qoa_decode_frame(ofs_src, p_qoa->frame_len, &p_qoa->desc, p_qoa->dec.ptr(), &p_qoa->dec_len);
 						}
 
-						uint32_t dec_idx = (interp_pos % QOA_FRAME_LEN) * p_qoa->desc.channels;
+						uint32_t dec_idx = interp_pos % QOA_FRAME_LEN << (is_stereo ? 1 : 0);
```

Comment:
I don't think readability is improved by this. And in terms of performance, I'm not even sure you gain anything. In this expression, the integer modulo is the most expensive operation, an integer multiplication is a 1-2 cycles operations, not more expensive than a condition and shift, I would say (maybe that could vary a little more on some platform). Anyway, trying to measure that ( on any platform), I don't think we can see any difference between the two.

Reply:
This is a template, so the condition doesn't exist after compilation.

Reply:
That makes more sense then. Still probably not any observable cycle difference.